addons/pos_loyalty/static/src/app/components/popups/manage_giftcard_popup/manage_giftcard_popup.js

```diff
@@ -21,19 +23,52 @@ export class ManageGiftCardPopup extends Component {
     };
 
     setup() {
+        this.pos = usePos();
         this.ui = useService("ui");
         this.state = useState({
             inputValue: this.props.startingValue,
             amountValue: "",
             error: false,
             amountError: false,
             expirationDate: luxon.DateTime.now().plus({ year: 1 }),
+            existingGiftCardId: false,
         });
         this.inputRef = useRef("input");
         this.amountInputRef = useRef("amountInput");
         onMounted(this.onMounted);
     }
 
+    async handleCodeInputEvents(event) {
+        clearTimeout(this.timeout);
+        this.timeout = setTimeout(async () => {
+            if (!this.inputRef.el) {
+                this.state.existingGiftCardId = false;
+                return;
+            }
+            this.state.existingGiftCardId = false;
+            const [loyaltyCard] = await this.pos.getGiftCard(this.inputRef.el.value.trim());
+            if (!loyaltyCard) {
+                return;
+            }
+            const linkedOrderIds = await this.pos.data.call(
+                "loyalty.card",
+                "get_loyalty_card_linked_orders",
+                [loyaltyCard.id]
+            );
```

Comment:
Can you create a method that will handle the verification and the error message? Like that we can make only 1 RPC